tools/integration_tests/implicit_dir/delete_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Provide test for deleting implicit directory.
+package implicit_dir_test
+
+import (
+	"path"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/operations"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup/implicit_and_explicit_dir_setup"
+)
+
+// Directory Structure
+// testBucket/implicitDirectory                                                  -- Dir
+// testBucket/implicitDirectory/fileInImplicitDir1                               -- File
+// testBucket/implicitDirectory/implicitSubDirectory                             -- Dir
+// testBucket/implicitDirectory/implicitSubDirectory/fileInImplicitDir2          -- File
+func TestDeleteNonEmptyImplicitDir(t *testing.T) {
+	// Clean the mountedDirectory before running test.
+	setup.CleanMntDir()
+
+	implicit_and_explicit_dir_setup.CreateImplicitDirectory()
+
+	dirPath := path.Join(setup.MntDir(), implicit_and_explicit_dir_setup.ImplicitDirectory)
+
+	implicit_and_explicit_dir_setup.RemoveAndCheckIfDirIsDeleted(dirPath, implicit_and_explicit_dir_setup.ImplicitDirectory, t)
+}
+
+// Directory Structure
+// testBucket/implicitDirectory                                                  -- Dir
+// testBucket/implicitDirectory/fileInImplicitDir1                               -- File
+// testBucket/implicitDirectory/implicitSubDirectory                             -- Dir
+// testBucket/implicitDirectory/implicitSubDirectory/fileInImplicitDir2          -- File
+func TestDeleteNonEmptyImplicitSubDir(t *testing.T) {
+	// Clean the mountedDirectory before running test.
+	setup.CleanMntDir()
+
+	implicit_and_explicit_dir_setup.CreateImplicitDirectory()
+
+	subDirPath := path.Join(setup.MntDir(), implicit_and_explicit_dir_setup.ImplicitDirectory, implicit_and_explicit_dir_setup.ImplicitSubDirectory)
```

Comment:
How are we creating implicit subDir?

Reply:
[This](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/tools/integration_tests/util/setup/implicit_and_explicit_dir_setup/testdata/create_objects.sh) script is creating whole structure.

Reply:
Okay, got it. Then this name (CreateImplicitDirectory) is misleading, feels like one empty implicit directory. Instead, we should rename it to CreateImplicitDirectoryStructure. 

Similarly for CreateExplicitDirectoryStructure.